lib/Store.lua

```diff
@@ -48,12 +48,18 @@ function Store.new(reducer, initialState, middlewares)
 	table.insert(self._connections, connection)
 
 	if middlewares then
-		local dispatch = Store.dispatch
+		local unboundDispatch = self.dispatch
+		local dispatch = function(...)
+			return unboundDispatch(self, ...)
+		end
+
 		for _, middleware in ipairs(middlewares) do
```

Comment:
They're not very intuitive, but at the same time I'm not sure there's a nicer way of specifying middleware order besides an array >.<

I think if we're going to stick to an array for middlewares then invoking them from the front of the list forward makes the most sense, but that's just me

Reply:
I agree that it's not intuitive either way.

I think the best approach is to probably document the way that the middleware is applied, because if you ever write any middleware there's a good chance you'll need to know one way or another.  As long as the info is easy to find that should be enough.